### PR TITLE
spelling fixes (round 1)

### PR DESCRIPTION
**Pull request by jsoref**
(see commit message for list of changes)

### CODE REVIEW

File: src/compiler/checker.ts [hunk 1]
```diff
@@ -3265,9 +3265,9 @@ namespace ts {
                 const outerTypeParameters = getOuterTypeParametersOfClassOrInterface(symbol);
                 const localTypeParameters = getLocalTypeParametersOfClassOrInterfaceOrTypeAlias(symbol);
                 // A class or interface is generic if it has type parameters or a "this" type. We always give classes a "this" type
-                // because it is not feasible to analyze all members to determine if the "this" type escapes the class (in particular,
+                // because it is not feasible to analyse all members to determine if the "this" type escapes the class (in particular,
```
Comment (RyanCavanaugh) [inline src/compiler/checker.ts]: This project is run from the United States so `analyze` is correct :wink:

**Comment by RyanCavanaugh** [comment]
Looks like you'll have to also run some of the fixes in `types.ts` so that this compiles

**Comment by jsoref** [comment]
@RyanCavanaugh: well, this is just the changes for the file itself.

I can update the request to include the corresponding fixes to the output, but I think it's probably best to start with you guys just reviewing the intentional changes before I worry about updating to include the test output.

File: src/compiler/checker.ts (hunk 1 above)
Comment (jsoref) [inline src/compiler/checker.ts]: I normally would use `z`, but there was more instances of `s`, so I favored it against my own preference. I'll gladly switch :-)

**Comment by jsoref** [comment]
Offhand, it looks like all of the changes in types.ts are either related or comments, tentatively, I'll include the whole file.

**Comment by RyanCavanaugh** [comment]
You'll have to take care of the downstream effects as well.

If `jake local` doesn't succeed, we can't accept the PR. Please see contributing.md.

File: src/compiler/types.ts [hunk 2]
```diff
@@ -1559,7 +1559,7 @@ namespace ts {
         /* @internal */ classifiableNames?: Map<string>;
         // Stores a mapping 'external module reference text' -> 'resolved file name' | undefined
         // It is used to resolve module names in the checker.
-        // Content of this fiels should never be used directly - use getResolvedModuleFileName/setResolvedModuleFileName functions instead
+        // Content of this fields should never be used directly - use getResolvedModuleFileName/setResolvedModuleFileName functions instead
```
Comment (sandersn) [inline src/compiler/types.ts]: should just be 'field'

**Comment by sandersn** [comment]
i don't see 'analyse' in the diff. Why is it on the list of corrections in the PR description?

**Comment by DanielRosenwasser** [comment]
@sandersn the correction was from "analyze" to "analyse", but was reverted.

**Comment by sandersn** [comment]
So it was in the previous PR? There's only one commit on this one.

**Comment by RyanCavanaugh** [comment]
Presumably there was a force push

**Comment by jsoref** [comment]
@RyanCavanaugh : the tests now pass, hopefully this is closer to what you want.

**Comment by sandersn** [comment]
@jsoref can you keep commits in the future so I can diff before vs after? It makes reviewing much easier.

**Comment by sandersn** [comment]
Looks good to me except for the [incorrect fix "fields"](https://github.com/Microsoft/TypeScript/pull/6932/files#diff-4b8bd1eea29904f1be39cd864e1a45c0R1562)

**Comment by yuit** [comment]
:+1:

**Comment by jsoref** [comment]
@sandersn : I've preserved it as https://github.com/jsoref/TypeScript/tree/spelling-1-1 which should hopefully keep it from going away.

**Comment by sandersn** [comment]
:+1:

**Comment by DanielRosenwasser** [comment]
Awesome - thanks @jsoref!

**Comment by jsoref** [comment]
@sandersn : I presume I can now delete my spelling-1-1 branch?

My next step is to try to repeat this process with a new spelling branch https://github.com/Microsoft/TypeScript/pull/6988

**Comment by sandersn** [comment]
Yep, go ahead. Looking forward to seeing more spelling fixes. :)

File: src/compiler/checker.ts [hunk 3]
```diff
@@ -16718,7 +16718,7 @@ namespace ts {
                     }
                 });
 
-                // ECMA-262 11.1.5 Object Initialiser
+                // ECMA-262 11.1.5 Object Initializer
```
Comment (mihailik) [inline src/compiler/checker.ts]: Is it 11.1.5?

It's not, according to the [ECMA web site](http://www.ecma-international.org/ecma-262/6.0/index.html#sec-object-initializer)

> 12.2.6 Object Initializer
> NOTE 1 An object initializer is an expression describing the initialization of an Object, written in a form resembling a literal. It is a list of zero or more pairs of property keys and associated values, enclosed in curly brackets. The values need not be literals; they are evaluated each time the object initializer is evaluated.

Comment (jsoref) [inline src/compiler/checker.ts]: Happy to fix that, but it's not a spelling mistake. And amusingly, I've at least (accidentally) fixed the spelling to match the spec... (I wonder how it got out of sync...)

Comment (jsoref) [inline src/compiler/checker.ts]: Filed #7009

**Pull request merged**